### DISCUSSION TITLE
How to get cpu boost clock speed on windows?

### DISCUSSION THREAD

**Discussion by Meldryt** [lfreist/hwinfo#88] [Q&A]
Currently Win32_Processor query functions gives only base and max clock speed.
For my 5800X3D it shows 3.4.Ghz base clock speed in both cases.
I want to get my max turbo clock speed, which is 4.5 Ghz i guess.

```
hr = obj->Get(L"MaxClockSpeed", 0, &vt_prop, nullptr, nullptr);
  if (SUCCEEDED(hr) && (V_VT(&vt_prop) == VT_I4)) {
    cpu._maxClockSpeed_MHz = vt_prop.uintVal;
    cpu._regularClockSpeed_MHz = vt_prop.uintVal;
  }
```
Is there any way to get the turbo clock speed with windows api?

**Comment by lfreist** [comment]
Hi, afaik WMI does not provide any interface for CPU boost frequency.

But I'll check if there are other ways to get boost frequency
